hazelcast/src/main/java/com/hazelcast/client/cache/impl/nearcache/NearCachedClientCacheProxy.java

```diff
@@ -324,7 +325,7 @@ protected CompletableFuture<Boolean> callRemoveAsync(K key, Data keyData, Data o
             } finally {
                 invalidateNearCache(toNearCacheKey(key, keyData));
             }
-        });
+        }, getDefaultAsyncExecutor());
```

Comment:
Should client-side near-cache upkeep run in the `FJP#commonPool`? 
I think it's best to move it to an executor separate from user code. In `NearCachedClientMapProxy` I see we use the `HazelcastClientInstanceImpl#getTaskScheduler()`, maybe that's a suitable executor?
cc @mdumandag

Reply:
Having FJP#commonPool seems ok to me since it is auto scalable and user code initiates these parts.

Reply:
We do have not a lot of options here: `FJP#commonPool` or `HazelcastClientInstanceImpl#getTaskScheduler()`